### ISSUE TITLE
Crash due to check-fail in CropAndResize

### ISSUE THREAD

**Issue by shijy16** [tensorflow/tensorflow#58177]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.12.0-dev20221018

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.8

### CUDA/cuDNN version

CUDA 11.5

### Current Behaviour?

```shell
A check-fail can be triggered in CropAndResize.
```

### Standalone code to reproduce the issue

```shell
import os
os.environ['TF_ENABLE_ONEDNN_OPTS'] = '0'
import tensorflow as tf
import numpy as np
print(tf.__version__)
for _ in range(20):
    try:
        with tf.device("GPU:0"):
            method = "bilinear"
            extrapolation_value = -90.71406992626788
            image = tf.saturate_cast(tf.random.uniform([1, 4, 1, 8], minval=0, maxval=64, dtype=tf.int64), dtype=tf.int64)
            boxes = tf.random.uniform([4, 0], dtype=tf.float32)
            box_ind = tf.saturate_cast(tf.random.uniform([8, 0, 10, 4], minval=0, maxval=64, dtype=tf.int64), dtype=tf.int32)
            crop_size = tf.saturate_cast(tf.random.uniform([2], minval=0, maxval=64, dtype=tf.int64), dtype=tf.int32)
            res = tf.raw_ops.CropAndResize(
                method=method,
                extrapolation_value=extrapolation_value,
                image=image,
                boxes=boxes,
                box_ind=box_ind,
                crop_size=crop_size,
            )
    except:
        pass
```

### Relevant log output

```shell
F tensorflow/core/framework/tensor_shape.cc:45] Check failed: NDIMS == dims() (1 vs. 4)Asking for tensor of 1 dimensions from a tensor of 4 dimensions
```
</details>

**Comment by tilakrayal** [comment]
@shijy16,
I tried to execute the code with the alternative approach by providing the `GPU name for the tf.device` and executed without any error/issues. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/0e7bcb4f1c678a4dc9ff076e705915cb/58177.ipynb) and also **tf.device** function specifies the device to be used for ops created/executed in a particular context. Nested contexts will inherit and also create/execute their ops on the specified device. 

If a specific device is not required, consider not using this function so that a device can be automatically assigned. **In general the use of this function is optional. device_name can be fully specified, as in "/job:worker/task:1/device:cpu:0", or partially specified,** containing only a subset of the "/"-separated fields.  Please take a look at this doc [link1](https://www.tensorflow.org/api_docs/python/tf/device) and [link2](https://www.tensorflow.org/guide/gpu#overview) for the reference. Thank you!

**Comment by shijy16** [comment]
@tilakrayal 
Thanks for your kindly reponse!
But I am still confused why `tf.device('GPU:0')` and `tf.device('CPU')` doesn't work for only this API. 
And once I remove the `tf.device` line and let it choose devices automatically, the bug shows up again.

**Comment by tilakrayal** [comment]
@shijy16,
Probably there is a missing check in the low level API.
Could you please try to reproduce your experiment with the high level API and let us know if you are facing same issue. Thank you!

**Comment by mihaimaruseac** [comment]
This is a real issue that used to be considered a vulnerability but due to large number of occurences versus limited impact it was decided to only consider it a bug. Please don't autoclose these types of issues.

However, please do check if these are happening in tf-nightly. If they don't, then the issue can be closed.

**Comment by tilakrayal** [comment]
@shijy16,
I tried to execute the code on tf-nightly and it was executed without any issue/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/0244a7b536e22168cb896bbfddb95d74/untitled732.ipynb) and let us know if this serves your requirement. Thank you!

**Comment by shijy16** [comment]
> @shijy16, I tried to execute the code on tf-nightly and it was executed without any issue/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/0244a7b536e22168cb896bbfddb95d74/untitled732.ipynb) and let us know if this serves your requirement. Thank you!

It seems to be fixed by PR #57783 for a similar issue #57754 , I'll close this issue while it is fixed recently.
